crypto/openssl/openssl_key_exchange_ecdh.c

```diff
@@ -468,14 +457,18 @@ int32_t crypto_hal_set_peer_random(void *context,
 		goto error;
 	}
 
-	key = key_ex_data->_key;
-	group = EC_KEY_get0_group(key);
+	group = EC_GROUP_new_by_curve_name(key_ex_data->group_name_nid);
+	if (group == NULL)
+	{
+		LOG(LOG_ERROR, "Failed to get the EC group\n");
+		goto error;
+	}
 	point = EC_POINT_new(group);
-	if (group == NULL || point == NULL || key == NULL) {
+	if (group == NULL || point == NULL) {
 		LOG(LOG_ERROR, "Error curve parameters are NULL\n");
 		goto error;
 	}
-	EC_POINT_set_affine_coordinates_GFp(group, point, Ax_bn, Ay_bn, ctx);
+	EC_POINT_set_affine_coordinates(group, point, Ax_bn, Ay_bn, ctx);
 	shx = fdo_alloc(bn_num_bytes(Ax_bn));
 	if (!shx) {
```

Comment:
Why shx is not freed ?

Reply:
shx is freed at line no.572 on right side(current code). We also used Valgrind linux tool to ensure no memory leaks while running this binary.